src/game/game.c

```diff
@@ -72,9 +72,15 @@ int32_t StopGame(void)
         return GF_EXIT_TO_TITLE;
     }
 
-    if (g_InvExtraData[0] == 0) {
-        return GF_START_SAVED_GAME | g_InvExtraData[1];
-    } else if (g_InvExtraData[0] == 1) {
+    if (g_InvExtraData[IED_PAGE_NUM] == 0
+        && g_InvExtraData[IED_PASSPORT_MODE] == PASSPORT_MODE_SHOW_SAVES) {
+        return GF_START_SAVED_GAME | g_InvExtraData[IED_SAVEGAME_NUM];
+    } else if (
+        g_InvExtraData[IED_PAGE_NUM] == 0
+        && g_InvExtraData[IED_PASSPORT_MODE] == PASSPORT_MODE_SELECT_LEVEL) {
+        Savegame_Load(g_InvExtraData[IED_SAVEGAME_NUM], &g_GameInfo);
```

Comment:
Is this the proper way to load the save game? I need to load the `g_GameInfo` from the slot the user is level selecting so that the proper `START_INFO` is used. Also this logic is repeated in different parts of `inventry.c` but I'm trying to minimize refactors on this branch.

Reply:
Yes, it is. It used to consist of two steps, but with the refactors from the last month it is now reduced to just 1.